core/store/src/flat_state.rs

```diff
@@ -267,6 +267,11 @@ impl FlatStateDelta {
         self.0.get(key).cloned()
     }
 
+    /// Merge this delta with other one. New changes should override the old ones.
```

Comment:
Can you be more specific for what's new or old? Maybe use other and self instead of old and new.